torch/_meta_registrations.py

```diff
@@ -135,3 +125,46 @@ def meta_adaptive_avg_pool2d(self, output_size):
 def meta_adaptive_avg_pool3d(self, output_size):
     check(self.ndim == 4 or self.ndim == 5, f"Expected 4D or 5D tensor, but got {self.shape}")
     return self.new_empty(self.shape[:-3] + tuple(output_size))
+
+@torch.library.impl(meta_lib, "index.Tensor")
+def meta_index_Tensor(self, indices):
+    # aten::index is the internal advanced indexing implementation
+    check(len(indices) <= self.ndim, f"too many indices for tensor of dimension {self.ndim} (got {len(indices)})")
+    # checkIndexTensorTypes
+    result = []
+    for i, index in enumerate(indices):
+        if index is not None:
+            check(index.dtype in [torch.long, torch.int8, torch.bool], "tensors used as indices must be long, byte or bool tensors")
+            if index.dtype in [torch.int8, torch.bool]:
+                nonzero = index.nonzero()
```

Comment:
hm, so it's not very meta then for bool indices?

Reply:
yup